02_knight_and_princess/kirill_tolstobrov/main.cpp

```diff
@@ -0,0 +1,165 @@
+#include <cstdlib>
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <utility>
+#include <vector>
+
+class Vertex {
+  int id;
+
+ public:
+  explicit Vertex(int init_id = 0) : id(init_id) {}
+  int get_id() const { return id; }
+};
+
+class Edge {
+  int id;
+  int vertex1_id;
+  int vertex2_id;
+
+ public:
+  Edge(int init_id, int v1_init, int v2_init)
+      : id(init_id), vertex1_id(v1_init), vertex2_id(v2_init){};
+  int get_id() const { return id; }
+  int get_v1_id() const { return vertex1_id; }
+  int get_v2_id() const { return vertex2_id; }
+};
+
+class Graph {
+  std::vector<Edge> edges;
+  std::vector<Vertex> vertices;
+
+  // connections_map: vertex1 -> edge -> vertex2
+  std::map<int, std::map<int, int>> connections_map;
+
+  int vertices_amount;
+  int edges_amount;
+
+ public:
+  Graph() : vertices_amount(0), edges_amount(0) {}
+
+  int get_vertices_amount() const { return vertices_amount; }
+  int get_edges_amount() const { return edges_amount; }
```

Comment:
Зачем для этого отдельные переменные, у тебя же уже есть:
```suggestion
  int get_vertices_amount() const { return vertices.size(); }
  int get_edges_amount() const { return edges.size(); }
```